core/src/main/java/org/apache/accumulo/core/metadata/ReferencedTabletFile.java

```diff
@@ -186,48 +195,35 @@ public String getNormalizedPathStr() {
     return parts.getNormalizedPath();
   }
 
-  /**
-   * Return a string for inserting a new tablet file.
-   */
-  public String getMetaInsert() {
-    return parts.getNormalizedPath();
-  }
-
-  /**
-   * Return a new Text object of {@link #getMetaInsert()}
-   */
-  public Text getMetaInsertText() {
-    return new Text(getMetaInsert());
-  }
-
   /**
    * New file was written to metadata so return a StoredTabletFile
    */
   public StoredTabletFile insert() {
-    return new StoredTabletFile(parts.getNormalizedPath());
+    return StoredTabletFile.of(getPath(), getRange());
   }
 
   @Override
   public int compareTo(ReferencedTabletFile o) {
     if (equals(o)) {
       return 0;
     } else {
-      return parts.getNormalizedPath().compareTo(o.parts.getNormalizedPath());
+      return comparator.compare(this, o);
     }
```

Comment:
This was an existing pattern in the code, may be better to skip the equals call.  In the not equals case it will read the data twice.

Reply:
Good point, i can drop the equals for sure, just an oversight, good catch. It should just be:

```java
  @Override
  public int compareTo(ReferencedTabletFile o) {
    return comparator.compare(this, o);
  }
```